cocos/animation/marionette/pose-graph/instantiation.ts

```diff
@@ -7,17 +7,51 @@ import { PoseGraph } from './pose-graph';
 import { PureValueNode, PureValueNodeLinkContext } from './pure-value-node';
 import { NodeInputPath } from './foundation/node-shell';
 import { PoseGraphNode } from './foundation/pose-graph-node';
+import {
+    AnimationGraphBindingContext, AnimationGraphSettleContext,
+    AnimationGraphUpdateContext, AnimationGraphEvaluationContext,
+} from '../animation-graph-context';
 
 type EvaluatableNode = PoseNode | PureValueNode;
 
 function isEvaluatableNode (node: PoseGraphNode): node is EvaluatableNode {
     return (node instanceof PoseNode || node instanceof PureValueNode);
 }
 
+class InstantiatedPoseGraph {
+    constructor (
+        private _rootPoseNode: PoseNode | undefined,
+    ) {
+
+    }
+
+    public bind (context: AnimationGraphBindingContext) {
+        this._rootPoseNode?.bind(context);
+    }
+
+    public settle (context: AnimationGraphSettleContext) {
+        this._rootPoseNode?.settle(context);
+    }
+
+    public reenter () {
+        this._rootPoseNode?.reenter();
+    }
+
+    public update (context: AnimationGraphUpdateContext) {
+        this._rootPoseNode?.update(context);
+    }
+
+    public evaluate (context: AnimationGraphEvaluationContext) {
+        return this._rootPoseNode?.evaluate(context) ?? null;
+    }
+}
+
+export type { InstantiatedPoseGraph };
+
 export function instantiatePoseGraph (
     graph: PoseGraph,
     linkContext: PureValueNodeLinkContext,
-): PoseNode | undefined {
+): InstantiatedPoseGraph {
```

Comment:
Before, the instantiation result is the instantiated root node. In later, extra information about the graph should be passed. So here should return an "instantiated graph".